asap/rough_align/fit_multiple_solves.py

```diff
@@ -0,0 +1,477 @@
+        r_tps = renderapi.connect(**self.args['thin_plate_output_stack'])
+        if self.args['translation_output_stack'] is not None:
+            r_trans = renderapi.connect(**self.args['translation_output_stack'])
+            do_translate = True
+        else:
+            do_translate = False
+        allZ = [int(z) for z in renderapi.stack.get_z_values_for_stack(
+            self.args['input_stack']['name'], render=r_in)]
+        
+        minZ = (allZ[0] if self.args["minZ"] is None else max(self.args["minZ"],
+                allZ[0]))
+       
+        maxZ = (allZ[-1] if self.args["maxZ"] is None else min(self.args["maxZ"],
+                allZ[-1]))
+
+        sol_range = (minZ, maxZ)
+
+        in_default = create_input_defaults(
+            render=r_in, pm_render=r_pm,
+            input_stack=self.args['input_stack']['name'],
+            pm_collection=self.args['pointmatch_collection']['name'],
+            solve_range=sol_range, pool_size=self.args['pool_size'],
+            multiz=False
+        )
+
+        '''---------------------Rigid solve-------------------------'''
+        rigid_result_rts = self.apply_transform(
+            new_matches=in_default['new_matches'],
+            rts=in_default['combined_tilespecs'], app='rigid')
+        self.save_transform(render=r_rot, rts=rigid_result_rts, app='rigid')
+        if do_translate:
+            '''---------------Translation solve -------'''
+            
+            trans_result_rts = self.apply_transform(
+                new_matches=in_default['new_matches'],
+                rts=rigid_result_rts, app='translate')
+            
+            self.save_transform(render=r_trans, rts=trans_result_rts,
+                                app='translate')
+            
+            '''---------------Affine solve ------------'''
+            
+            aff_result_rts = self.apply_transform(
+                new_matches=in_default['new_matches'],
+                rts=trans_result_rts, app='affine')
+        else:
+            '''---------------Affine solve ------------'''
+
+            aff_result_rts = self.apply_transform(
+                new_matches=in_default['new_matches'],
+                rts=rigid_result_rts, app='affine')
+        self.save_transform(render=r_aff, rts=aff_result_rts, app='affine')
+            
+        '''---------------Thin_plate_spline solve--------------------'''
+        tps_result_rts = self.apply_transform(
+            new_matches=in_default['new_matches'],
+            rts=aff_result_rts, app='tps')
+        self.save_transform(render=r_tps, rts=tps_result_rts, app='tps')
+        
+        allZ_out = [int(z) for z in renderapi.stack.get_z_values_for_stack(
+            self.args["rigid_output_stack"]['name'],
+            render=renderapi.connect(**self.args["rigid_output_stack"]))]
+
+        out_dict = {
+            'zs': allZ_out,
+            'rigid_output_stack': self.args['rigid_output_stack']['name'],
```

Comment:
These were changed to `output_stack` schemas but still just giving them as strings here.